process/asyncExecution/headersExecutor_test.go

```diff
@@ -0,0 +1,103 @@
+package asyncExecution
+
+import (
+	"testing"
+	"time"
+
+	"github.com/multiversx/mx-chain-core-go/data"
+	"github.com/multiversx/mx-chain-core-go/data/block"
+	"github.com/multiversx/mx-chain-go/process/asyncExecution/queue"
+	"github.com/multiversx/mx-chain-go/testscommon/processMocks"
+	"github.com/stretchr/testify/require"
+)
+
+func createMockArgs() ArgsHeadersExecutor {
+	headerQueue, _ := queue.NewBlocksQueue()
+
+	return ArgsHeadersExecutor{
+		BlocksQueue:      headerQueue,
+		ExecutionTracker: &processMocks.ExecutionTrackerStub{},
+		BlockProcessor:   &processMocks.BlockProcessorStub{},
+	}
+}
+
+func TestNewHeadersExecutor(t *testing.T) {
+	t.Parallel()
+
+	t.Run("nil headers queue", func(t *testing.T) {
+		args := createMockArgs()
+		args.BlocksQueue = nil
+
+		_, err := NewHeadersExecutor(args)
+		require.Equal(t, ErrNilHeadersQueue, err)
+	})
+
+	t.Run("nil execution tracker", func(t *testing.T) {
+		args := createMockArgs()
+		args.ExecutionTracker = nil
+
+		_, err := NewHeadersExecutor(args)
+		require.Equal(t, ErrNilExecutionTracker, err)
+	})
+
+	t.Run("nil block processor", func(t *testing.T) {
+		args := createMockArgs()
+		args.BlockProcessor = nil
+
+		_, err := NewHeadersExecutor(args)
+		require.Equal(t, ErrNilBlockProcessor, err)
+	})
+
+	t.Run("should work", func(t *testing.T) {
+		args := createMockArgs()
+
+		executor, err := NewHeadersExecutor(args)
+		require.NoError(t, err)
+		require.NotNil(t, executor)
+		require.False(t, executor.IsInterfaceNil())
+
+		err = executor.Close()
+		require.NoError(t, err)
+	})
+}
+
+func TestHeadersExecutor_StartAndClose(t *testing.T) {
```

Comment:
can you try several scenarios here?
e.g. 
- processing fails with replacement block -> should not retry
- processing fails without replacement block (no other block in the queue) -> should retry
- processing fails with a block in the queue but for next nonce (so we cannot bypass an error here) -> should retry